pkg/vault/contracts/BatchRouter.sol

```diff
@@ -951,7 +961,18 @@ contract BatchRouter is IBatchRouter, BatchRouterStorage, RouterCommon, Reentran
             // Treat all ERC4626 pool tokens as wrapped. The next step will verify if we can use the wrappedToken as
             // a valid ERC4626.
             IERC4626 wrappedToken = IERC4626(address(erc4626PoolTokens[i]));
-            IERC20 underlyingToken = IERC20(wrappedToken.asset());
+            IERC20 underlyingToken = IERC20(_vault.getBufferAsset(wrappedToken));
+
+            // If the vault returns address 0 as underlying, it means that the ERC4626 token buffer was not
+            // initialized. Thus, the router treats it as a non-ERC4626 token.
+            if (address(underlyingToken) == address(0)) {
+                if (isStaticCall == false) {
+                    underlyingAmounts[i] = amountsIn[i];
+                    wrappedAmounts[i] = amountsIn[i];
```

Comment:
Hmm don't we need to assign these two outside `if (isStaticCall == false)`?

Reply:
In a static call `underlyingAmounts[i]` and `wrappedAmounts[i]` will be 0.
In `addLiquidityERC4626PoolUnbalancedHook` we use those as max input amounts to add liquidity.

I don't get why no tests are failing because of this, since the query would fail when trying to add liquidity with 0 max inputs here: https://github.com/balancer/balancer-v3-monorepo/pull/892/files#diff-8c740872f6bae09a05c4cdf1ba7a2c0cae6ca40cf8ae0b14058d20e624fb14deR195

Reply:
Good catch! I created query tests for this now.